### PR TITLE
Proposal: default virtualization storage class

### PR DESCRIPTION
**Pull request by akalenyu**
This design presents a possible way of steering kubevirt users
towards using a specific set of backend storage parameters for provisioning VM disks

### CODE REVIEW

**Comment by akalenyu** [comment]
/cc @lyarwood - Do you happen to know if we have any places where a missing storage class gets computed wrt to instancetypes?
I assume we create DVs and delegate the storage class choices to CDI?
/cc @mhenriks

**Comment by lyarwood** [comment]
> /cc @lyarwood - Do you happen to know if we have any places where a missing storage class gets computed wrt to instancetypes? I assume we create DVs and delegate the storage class choices to CDI? /cc @mhenriks

@akalenyu We provide the `preference.spec.volumes.preferredStorageClassName` attribute to allow a preference to express a preferred storage class if one isn't provided for a given `Volume` within the `VirtualMachine` already but it's not used by any of our default `common-instancetype` preferences at the moment nor will it ever be tbh.

https://kubevirt.io/api-reference/main/definitions.html#_v1beta1_volumepreferences

**Comment by mhenriks** [comment]
I think this is a simple no nonsense way to allow for setting a default storage class for KubeVirt VMs.

It may be somewhat confusing to users who don't know this annotation exists and are expecting the default cluster storage class to be used.  So maybe emit an event when this KubeVirt default is chosen?

File: design-proposals/default-virt-storage-class.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Scalability
+It is assumed scalability was one of the choosing factors when identifying a storage class to be superior for VM workloads
+
+## Update/Rollback Compatibility
+* Existing populated PVCs will not be affected, however,  
+clone sources from previous versions may have been provisioned using a different storage class variation of a provisioner.
+* Storage classes can still be explicitly requested on manifests to override virt/k8s default storage class
+
+## Functional Testing Approach
+* Replicate I/O pattern that triggers CRC errors on Windows workloads
```
[the remark this answers is not in the document]
Comment (akalenyu) [reply design-proposals/default-virt-storage-class.md]: Yep makes sense, pushed a change to reflect these

**Comment by vladikr** [comment]
/approve
Thanks!

**Pull request merged**